cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -465,29 +469,74 @@ func (s *Server) Index(args *indexArgs) (state indexState, err error) {
 		return indexStateEmpty, createEmptyShard(args)
 	}
 
+	repositoryName := args.BuildOptions().RepositoryDescription.Name
+	if _, ok := s.deltaBuildRepositoriesAllowList[repositoryName]; ok {
+		repositoryID := args.BuildOptions().RepositoryDescription.ID
+		debug.Printf("Server.Index: marking %q (ID %d) for delta build", repositoryName, repositoryID)
+
+		args.UseDelta = true
+	}
```

Comment:
nit: create a helper function so the code here is less verbose

```go
args.UseDelta = s.UseDelta(args)
```

Reply:
I think this helper function would be used in exactly one place (here) is it still worth introducing it (adding a single-use function to Server's interface)? If we want to keep the log line, either the helper function would have a side effect (strange), or we'd need to keep most of this existing code in order to generate it.

Reply:
Its a matter of taste I suppose. I think it is worth it since it makes reading the code clearer.